### ISSUE TITLE
Transient DBContext no longer working correctly after Update to 3.0

### ISSUE THREAD

**Issue by PassionateDeveloper86** [dotnet/efcore#18144]
I was updating from 2.2 to 3.0 and using the DI with a transient DbContext like this:

```c#
services.AddDbContext<EloContext>(options => options.UseSqlServer(ConfigurationHelper.GetEloConString(config)), ServiceLifetime.Transient);
```

I am using DI within my constructor, e.g.:

```c#
using System.Linq;
using System.Threading.Tasks;
using Microsoft.EntityFrameworkCore;
using SWDi.Base.Models;
using SWDi.DataAccess.Intranet.AutoGenPOCOCode;
using SWDi.DataAccess.Intranet.Repository.Admin.UserModule.Mutation;

namespace SWDi.DataAccess.Intranet.Repository.Admin.UserModule.Query
{
    /// <inheritdoc />
    public class GetUserDataQuery : IGetUserDataQuery
    {
        /// <summary>
        /// Initializes a new instance of the <see cref="GetUserDataQuery"/> class.
        /// </summary>
        /// <param name="context">Intranet Context by DI.</param>
        public GetUserDataQuery(IntranetContext context)
        {
            this.Context = context;
        }

        private IntranetContext Context { get; }

        /// <summary>
        /// Get the Intranet user context using the users (AD) username.
        /// </summary>
        /// <param name="username">The user ID.</param>
        /// <returns>The DB user context.</returns>
        public async Task<User> Execute(string username)
        {
            User u = await this.Context.Users
                .Where(x => x.Username == username)
                .Include(x => x.RefDepartment)
                .Include(x => x.RefUserRole)
                .SingleOrDefaultAsync();

            return u;
        }
    }
}
```

On my frontpage (build with react) I do several requests on login to fetch the first needed, later cached, data - some of them are accessing the same table at the same time - on 2.2 perfectly fine. Since 3.0 I get an error:
"A second operation started on this context before a previous operation completed. This is usually caused by different threads using the same instance of DbContext. For more information on how to avoid threading issues with DbContext, see https://go.microsoft.com/fwlink/?linkid=2097913."

Through git I can pinpoint that to the moment of the 3.0 update.

**Comment by jsheetzati** [comment]
I am also running into a similar issue. Our application uses scoped contexts via `services.AddDbContext` without specifying a lifetime. However, we also initialize new DbContexts that are needed for atomic updates or happen in parallel. Each async operation happening in parallel has its own context. Something like this.

```csharp
public async Task SomeMethod()
{
    var tasks = new List<Task>(); 
    using (var context = CreateNewDbContext())
    {
        tasks.Add(DoSomething());
    }
    await Task.WhenAll(tasks);
}

private async Task DoSomething()
{
   using (var context = InitializeNewContext())
   {
      context.Add(new Book { Title = "title" });
      await context.SaveChangesAsync();
   }
}
```

We did not have any context threading issues in 2.2 and am having difficulty getting to the root of the problem.

**Comment by ajcvickers** [comment]
@Wasseratem @jsheetzati So far I have not been able to reproduce this. Can you please post a small, runnable project/solution that demonstrates the issue. This might require running things in a tight loop to force a concurrency issue.

**Comment by jsheetzati** [comment]
Also having difficulties reproducing in a sample application at this point. This is leading me to believe we have a different issue in our application :( 
Thanks @ajcvickers

**Comment by ajcvickers** [comment]
**EF Team Triage:** Closing this issue as the requested additional details have not been provided and we have been unable to reproduce it.

*BTW this is a canned response and may have info or details that do not directly apply to this particular issue. While we'd like to spend the time to uniquely address every incoming issue, we get a lot traffic on the EF projects and that is not practical. To ensure we maximize the time we have to work on fixing bugs, implementing new features, etc. we use canned responses for common triage decisions.*

**Comment by Wasseratem** [comment]
Still having this problem, I fought with this for 6 hours today without any idea how to solve it or even reproduce it in a test anvoiriment...

**Comment by Wasseratem** [comment]
![image](https://user-images.githubusercontent.com/20551244/67632082-ec085100-f89e-11e9-9abc-430d0c19027d.png)

I found something interesting.

I have 3 methods: a lock / an unlock and a get.

The lock writes the userId into a column, the unlock delets it and the get gets the actual data. I have caching problems here, which had lead me again the problem if the transistent context - because of this, as far as I understand, there should be not a single cache: the db context should generated freshly and newly every request I do. 

So the lock and the get are within the same WebAPI call, but with the DI the context should be new (2 different classes, db context is inserted in the constructor).
The unlock is a single WebAPI call.

I refreshed my page a few times and clicked on the buttons and see:
Every Lock, every unlock and every get will have the SAME context ID - what?
